components/director/internal/domain/application/service.go

```diff
@@ -301,6 +305,21 @@ func (s *service) Delete(ctx context.Context, id string) error {
 		return errors.Wrapf(err, "while loading tenant from context")
 	}
 
+	scenarios, runtimes, err := s.getScenariosAndRuntimes(ctx, appTenant, id)
```

Comment:
usually methods that contain  "and" are handling multiple responsibilities which is not advicable from clean code pov.  It's also making the code confusing. For example, right now just reading through this method im left under the impression that you are first fetching both scenarios and runtimes and if there are no scenarios we return. And this is not the case - i have to go into the getScenariosAndRuntimes method and see that even though we are saying we fetch both, we only fetch rts if scenarios exist. 

So split those into two methods